be/src/formats/orc/column_reader.cpp

```diff
@@ -592,42 +533,60 @@ Status StringColumnReader::get_next(orc::ColumnVectorBatch* cvb, ColumnPtr& col,
+            }
             if (_type.type == TYPE_CHAR) {
+                size_t write_pos = vb.size();
                 // Possibly there are some zero padding characters in value, we have to strip them off.
-                for (int i = col_start; i < col_start + size; ++i, ++pos) {
-                    nulls[i] = !cvb->notNull[pos];
+                for (size_t i = col_start; i < col_start + size; ++i, ++pos) {
                     if (cvb->notNull[pos]) {
                         size_t str_size = remove_trailing_spaces(data->data[pos], data->length[pos]);
-                        vb.insert(vb.end(), data->data[pos], data->data[pos] + str_size);
-                        vo.emplace_back(vb.size());
+                        strings::memcpy_inlined(&vb[write_pos], data->data[pos], str_size);
+                        write_pos += str_size;
+                        vo.emplace_back(write_pos);
                     } else {
-                        vo.emplace_back(vb.size());
+                        vo.emplace_back(write_pos);
                     }
                 }
+                // set the correct size for vb
+                vb.resize(write_pos);
             } else {
+                size_t write_pos = vb.size();
                 for (int i = col_start; i < col_start + size; ++i, ++pos) {
-                    nulls[i] = !cvb->notNull[pos];
                     if (cvb->notNull[pos]) {
-                        vb.insert(vb.end(), data->data[pos], data->data[pos] + data->length[pos]);
-                        vo.emplace_back(vb.size());
+                        strings::memcpy_inlined(&vb[write_pos], data->data[pos], data->length[pos]);
+                        write_pos += data->length[pos];
+                        vo.emplace_back(write_pos);
                     } else {
-                        vo.emplace_back(vb.size());
+                        vo.emplace_back(write_pos);
                     }
                 }
+                // set the correct size for vb
+                vb.resize(write_pos);
             }
         } else {
             if (_type.type == TYPE_CHAR) {
                 // Possibly there are some zero padding characters in value, we have to strip them off.
+                size_t write_pos = vb.size();
                 for (int i = col_start; i < col_start + size; ++i, ++pos) {
                     size_t str_size = remove_trailing_spaces(data->data[pos], data->length[pos]);
-                    vb.insert(vb.end(), data->data[pos], data->data[pos] + str_size);
-                    vo.emplace_back(vb.size());
+                    strings::memcpy_inlined(&vb[write_pos], data->data[pos], str_size);
+                    write_pos += str_size;
+                    vo.emplace_back(write_pos);
                 }
+                // set the correct size for vb
+                vb.resize(write_pos);
             } else {
+                size_t write_pos = vb.size();
```

Comment:
```
memcpy(vb.data, data->data, data->size);
```

Reply:
orc's StringVectorBatch structure:
```c++
// pointers to the start of each string
DataBuffer<char*> data;
// the length of each string
DataBuffer<int64_t> length;
// string blob
DataBuffer<char> blob;
// dict codes, iff. there is dictionary.
DataBuffer<int64_t> codes;
```
We can't make sure each line's string is continuous in `blob`.